src/NServiceBus8.0/XmlSerializerFacade.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using Common;
+using Common.Tests;
+using NServiceBus;
+using NServiceBus.MessageInterfaces.MessageMapper.Reflection;
+using NServiceBus.Serialization;
+using NServiceBus.Settings;
+using NServiceBus.Unicast.Messages;
+
+class XmlSerializerFacade : ISerializerFacade
+{
+    public XmlSerializerFacade(params Type[] objectTypes)
+    {
+        mapper = new MessageMapper();
+        var settings = new SettingsHolder();
+        var conventions = CreateTestConventions(settings);
+        // evil hack
+        settings.Set((MessageMetadataRegistry)Activator.CreateInstance
```

Comment:
Ugh, that looks ugly.

Reply:
yeah. I've copied that from the v6/v7 versions that are doing the same thing.